lib/minio/minio_client.rb

```diff
@@ -0,0 +1,175 @@
+    response.status
+  end
+
+  def admin_policy_list
+    send_request("GET", admin_uri("list-canned-policies"))
+  end
+
+  def admin_policy_add(policy_name, policy)
+    body = JSON.generate(policy).encode("UTF-8")
+    response = send_request("PUT", admin_uri("add-canned-policy?name=#{policy_name}"), body)
+    response.status
+  end
+
+  def admin_policy_info(policy_name)
+    send_request("GET", admin_uri("info-canned-policy?name=#{policy_name}"))
+  end
+
+  def create_bucket(bucket_name)
+    response = send_request("PUT", s3_uri(bucket_name))
+    response.status
+  end
+
+  def delete_bucket(bucket_name)
+    response = send_request("DELETE", s3_uri(bucket_name))
+    response.status
+  end
+
+  def bucket_exists?(bucket_name)
+    response = send_request("GET", s3_uri(bucket_name))
+
+    response.status == 200
+  end
+
+  def list_objects(bucket_name, folder_path, max_keys: 1000)
+    objects = []
+    response = send_request("GET", s3_uri("#{bucket_name}?delimiter=&encoding-type=url&list-type=2&prefix=#{folder_path}&max-keys=#{max_keys}"))
+    parsed_objects = parse_list_objects(response.data[:body])
+    objects.concat(parsed_objects[0])
+
+    is_truncated = parsed_objects[1]
+    continuation_token = parsed_objects[2]
+    while is_truncated
+      response = send_request("GET", s3_uri("#{bucket_name}?continuation-token=#{continuation_token}&delimiter=&encoding-type=url&list-type=2&prefix=#{folder_path}&max-keys=#{max_keys}&start-after=#{continuation_token}"))
+      parsed_objects = parse_list_objects(response.data[:body])
+      objects.concat(parsed_objects[0])
+      is_truncated = parsed_objects[1]
+      continuation_token = parsed_objects[2]
+    end
+
+    objects
+  end
+
+  def send_request(method, uri, body = nil)
+    headers = @signer.build_headers(method, uri, body, @creds, REGION)
+
+    full_path = uri.path + (uri.query ? "?" + uri.query : "")
+    response = @client.request(method: method, path: full_path, headers: headers, body: body)
+    if [200, 204, 206, 404].include?(response.status)
+      response
+    else
+      raise "Error: #{response.body}"
+    end
+  end
+
+  private
+
+  def parse_list_objects(response)
+    # Parse the XML response
+    doc = Nokogiri::XML(response)
+    bucket_name = doc.xpath("//xmlns:Name").text
+    encoding_type = doc.xpath("//xmlns:EncodingType").text
+    # Process 'Contents' elements
+    objects = doc.xpath("//xmlns:Contents").map do |node|
+      Object.from_xml(node, bucket_name, encoding_type: encoding_type)
+    end
+
+    # Note to future: We may need to process 'CommonPrefixes' elements
+    # when we implement new APIs.
+    is_truncated = doc.xpath("//xmlns:IsTruncated").text.casecmp("true").zero?
+    continuation_token = doc.xpath("//xmlns:NextContinuationToken").text
+    if continuation_token && encoding_type == "url"
+      continuation_token = CGI.escape(continuation_token)
+    end
+    [objects, is_truncated, continuation_token]
+  end
+
+  class Object
```

Comment:
Risky name for an identifier. It's slightly out of alignment, but perhaps calling it "Blob" would spare you more problems than it causes (in being different than the MinIO nomenclature)